src/Components/WaitingRoom.tsx

```diff
@@ -38,39 +50,123 @@ const WaitingRoom = ({isHost}: {isHost: boolean}) => {
     gameDispatch({type: "startGame"})
   }
 
-  const backToLobby = () => {
-    gameDispatch({ type: "exitRoom" })
+  const style = {
+    position: 'absolute' as 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: 400,
+    bgcolor: 'background.paper',
+    border: '2px solid #000',
+    boxShadow: 24,
+    p: 4,
+  };
+
+  const exitRoom = async () => {
+    if (players.length === 1) {
+      deleteRoom()
+    } else {
+      // ! Feel like there could be async issues below but need revision
+      // * remove current player from player list 
+      const currentPlayerIndex: number = players.findIndex((player: any) => player.number === currentPlayer.number)
```

Comment:
There is a nicer way you can do this. :)
line 71-74
const updatePlayers = players.filter(dbPlayer => dbPlayer.number !== player.number)

Reply:
![chef's kiss](https://www.the-sun.com/wp-content/uploads/sites/6/2022/12/EP_CHEFS_KISS_EXPL_COMP.jpg?w=1280&quality=44)

Reply:
oh actually i think this didn't work for me. `filter()` only provides a shallow copy, but the spread operator creates a deep copy which then destructively splices the player out of this version

Reply:
ahhh yep i made it a deep copy
`const updatePlayers = [...players].filter(dbPlayer => dbPlayer.number !== currentPlayer.number)`

Reply:
yea i see what you mean.
filter wont destructively change your original array. but you can save its value and use that, which will give you a new array of values.
so

```
const players = [{number: 1, number: 2 , number: 3}]
players.filter(player => player !== 2)

console.log(players) // --> [{number: 1, number: 2 , number: 3}]
```
 vs
```
const players = [{number: 1, number: 2 , number: 3}]
const updatedPlayers = players.filter(player => player !== 2)

console.log(players) // --> [{number: 1, number: 2 , number: 3}]
console.log(updatedPlayers) // --> [{number: 1, number: 3}]
```

plus
```
const players = [{number: 1, number: 2 , number: 3}]
const updatedPlayers = [...players]
// saving a copy of players in updatedPlayers and then doing destructive operations on newPlayers would be the same memory wise as
// Just saving a shallow copy of a filtered array into updatedPlayers
```

Reply:
ahhh yep you are right. I think i forgot to change your original line:
`const updatePlayers = players.filter(dbPlayer => dbPlayer.number !== player.number)`
from `player.number` to `currentPlayer.number`